storage/store/inmemory/unsynchronized/events.go

```diff
@@ -131,3 +132,20 @@ func (e *Events) BatchStore(flow.Identifier, []flow.EventsList, storage.ReaderBa
 func (e *Events) BatchRemoveByBlockID(flow.Identifier, storage.ReaderBatchWriter) error {
 	return fmt.Errorf("not implemented")
 }
+
+// AddToBatch adds all the in-memory storages to the given batch.
+// It is used for the batching writes to the DB.
+func (e *Events) AddToBatch(batch storage.ReaderBatchWriter) error {
+	writer := batch.Writer()
+
+	for blockID, events := range e.blockIdToEvents {
```

Comment:
We are duplicating the same logic here with storage/store/events.go#BatchStore.

What about the inmemory module holding a storage/store/events.go#Events module, so that adding to batch will call storage/store/events.go#BatchStore internally?

Alternatively, the inmemory will return the data of [blockID, blockEvents], and let the caller pass them to  storage/store/events.go#BatchStore method.

Both approaches can avoid duplicated logic here. 

Thoughts?

Reply:
This is a fair point. also, not using the storage methods means the caches will get out of sync.

I think the second option makes the most sense:
* Instead of `AddToBatch`, have a method that returns the data stored in the cache.
* The persister module can take the data plus the blockID from its `ExecutionResult` and call `BatchStore` directly on the storage object.

@illia-malachyn it probably makes sense to implement this in a separate PR.